src/Controller/OmniaController.php

```diff
@@ -129,6 +145,117 @@ public function videoList() {
     ];
   }
 
+  /**
+   * Retrieve video data field name.
+   *
+   * @return \string
+   *   The name of the field.
+   *
+   * @throws \Exception
+   */
+  protected function getVideoFieldName() {
+    $entity_type_id = 'media';
+    $videoBundle = $this->config('nexx_integration.settings')
+      ->get('video_bundle');
+
+    $fieldDefinitions = $this->entityFieldManager->getFieldDefinitions($entity_type_id, $videoBundle);
+    foreach ($fieldDefinitions as $fieldname => $fieldDefinition) {
+      if ($fieldDefinition->getType() === 'nexx_video_data') {
+        $videoField = $fieldname;
+        break;
+      }
+    }
+
+    if (empty($videoField)) {
+      throw new \Exception('No video data field defined');
+    }
+
+    return $videoField;
+  }
+
+  /**
+   * Check if token is valid.
+   *
+   * @return bool
+   *   Validity of token.
+   *
+   * @throws \Exception
+   */
+  protected function isTokenValid($request) {
+    $token = $request->query->get('token', NULL);
+
+    if ($token === NULL) {
+      $this->logger->error('Missing token in request');
+      throw new AccessDeniedHttpException();
+    }
+
+    $config = $this->config('nexx_integration.settings');
+    if ($token != $config->get('notification_access_key')) {
+      $this->logger->error('Wrong token in request');
+      throw new AccessDeniedHttpException();
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Return Nexx Id of video.
+   *
+   * @return null|int
+   *   Nexx Id of video.
+   */
+  protected function getVideoId() {
+    return $this->videoData->itemData->general->ID;
```

Comment:
ist it 100% sure that the property is available? Otherwise add an `isset()` check before please.